Several combined pull requests + new gem (colored2) + RSpec + new syntax

kigster opened the pull request:
I am fine if this PR will not be merged, as this is arguably a different gem (it actually is – `colored2`).

But since the main author's fork does not appear to be maintained anymore, I am willing to take over and take pull requests from folks that want to use this gem with more modern ruby and modern frameworks.

I rewrote all tests in RSpec, added some more, refactored the code and added new goodies and syntax. 

I am already using it in a couple of private projects.

Submisions welcome!

Commit:
Refactor, cleanup, add functionality

Commit:
Docs update

Commit:
Adding usage image

Commit:
Remove ruby 2.3 syntax

Commit:
Badge update

Commit:
Add test coverage

Commit:
Adding codeclimate configuration

Commit:
Working revision 2.0 with new features

- global on/off switch
- ability to decorate any class
- abllity to decorate Object space
- additional irb session
- additional tests
- refactored and simplified code path

Commit:
Add back lost authors

Commit:
Add original author back to the LICENSE

Commit:
Fix year

Commit:
Fixing homepage

kigster commented:
@defunkt Curious to know what your plans are for this little gem of a `gem` ;) 

If you have no time or desire maintaining it – I volunteer to take over :)

defunkt commented:
I think it'd be best to start a new gem and let this one rest in peace. Thanks!

kigster commented:
Sounds good to me.

If you were so kind as to point to this fork in the README, so that people don't waste time forking and re-adding things that are already there, that would be lovely – but certainly it's entirely up to you.

Thanks for writing this wonderfully simple gem (oh, and not to mention – GitHub :)

kigster committed:
Make it more object oriented

kigster committed:
Version 3.0.0; rename clear to no_color; add #to_bol

kigster committed:
Fixed gem name in README

orta committed:
Add ruby 2.0 to travis

orta committed:
Update simplecov - see https://github.com/codeclimate/ruby-test-reporter/blob/master/CHANGELOG.md#v100-2016-11-03

orta committed:
Make tests green

kigster committed:
Merge pull request #1 from orta/colored2ruby

Adds support for OS X system rubies

kigster committed:
Bump gem version to 3.0.1

kigster committed:
Gem version 3.1.0; ruby 2.4; move spec files.

kigster committed:
Avoiding circular references

kigster committed:
Adding code coverage; code climate

kigster committed:
Fixing badges

kigster committed:
Gitter

kigster committed:
Updating permissions

kigster committed:
Updating gitignore

bastelfreak committed:
Add Ruby 2.7 to Travis-CI matrix

Ruby 2.7 got released some time ago. The first distributions start
shipping it. That's why I think we should add it to test test matrix.

kigster committed:
Merge pull request #5 from bastelfreak/ruby

kigster committed:
Adding more rubies and removing 2.0.0

kigster committed:
Update README.md

kigster committed:
Update README.md

kigster committed:
Create gem-push.yml

kigster committed:
Update and rename gem-push.yml to ruby-ci.yml

kigster committed:
Updating colored2 for Ruby 3.2; rspecs, rubocop CI

kigster committed:
Merge pull request #7 from kigster/kig/colored-4.0.0-alpha

Updating colored2 for Ruby 3.2; rspecs, rubocop CI

kigster committed:
Adding build statuses

kigster committed:
Merge pull request #8 from kigster/kig/badges

Adding build statuses

kigster committed:
Bump version

kigster committed:
Merge pull request #9 from kigster/kig/version-4.0.0

Bump version

kigster committed:
Hope this fixes the permissions problem

kigster committed:
Updated rubocop workflow

kigster committed:
Merge pull request #12 from kigster/kig/candidate-4.0.1

Hope this fixes the permissions problem

kigster committed:
Adding Gemfile.lock to the repo

kigster committed:
Add rubocop task; make default run spec & rubocop
